math/check_factorial.cpp

```diff
@@ -0,0 +1,64 @@
+/**
+ * @file
+ * @brief A simple program to check if the given number is a factorial of some
+ * number or not.
+ * @author [Divyajyoti Ukirde](https://github.com/divyajyotiuk)
+ */
+#include <cassert>
+#include <iostream>
+
+/**
+ * Function to check if the given number is factorial of some number or not.
+ * @param n number to be checked.
+ * @return if number is a factorial, returns true, else false.
+ */
+
+bool is_factorial(uint64_t n) {
+    if (n <= 0) {
+        return false;
+    }
+    for (uint32_t i = 1;; i++) {
```

Comment:
Why uint32_t here and uint64_t in parameter of function. Can't both be uint64_t @kvedala ?

Reply:
sure, they can. the logic is that the factorial result will overflow the `uint64_t` much early on. the variable `i` would not even get to limit of uint8_t.

Reply:
I got your point.
but then variable  `i` should also be uint64_t

Reply:
it can be but doesn't have to be. :)